### PR TITLE
Licence update

### PR DESCRIPTION
**Pull request by ghost**
Licence added  to the newly created files as well as the modified ones.

### CODE REVIEW

File: compiler/src/main/java/io/github/landerlyoung/jenny/FieldIdDeclaration.kt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Copyright (C) 2024 The Qt Company Ltd.
```
Comment (villevoutilainen) [inline compiler/src/main/java/io/github/landerlyoung/jenny/FieldIdDeclaration.kt]: Shouldn't we have the license header in this file as well?

File: compiler/src/main/java/io/github/landerlyoung/jenny/JteData.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// Copyright (C) 2024 The Qt Company Ltd.
```
Comment (villevoutilainen) [inline compiler/src/main/java/io/github/landerlyoung/jenny/JteData.kt]: Here too, where's the license header?

File: compiler/src/main/java/io/github/landerlyoung/jenny/MethodIdDeclaration.kt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// Copyright (C) 2024 The Qt Company Ltd.
```
Comment (villevoutilainen) [inline compiler/src/main/java/io/github/landerlyoung/jenny/MethodIdDeclaration.kt]: Here too, license header?

File: sample-android/templates/constructor_definition.kte [hunk 4]
```diff
@@ -1,3 +1,7 @@
+<%--
```
Comment (villevoutilainen) [inline sample-android/templates/constructor_definition.kte:1]: License headers in these as well?

**Review by villevoutilainen — CHANGES_REQUESTED** [review]

File: sample-android/templates/initclass_preamble.kte [hunk 5]
```diff
@@ -7,21 +21,21 @@
 @param jteData: JteData
 ${jteData.initClassPrefix} bool ${jteData.className}::initClazz(JNIEnv* env) {
 #define JENNY_CHECK_NULL(val)                      \
-       do {                                        \
-           if ((val) == nullptr) {                 \
+do {                                        \
```
Comment (villevoutilainen) [inline sample-android/templates/initclass_preamble.kte]: Uhh.. please undo these changes, the indentation in this template is crafted to match the  built-in codegen 100%, including whitespace.

**Review by villevoutilainen — CHANGES_REQUESTED** [review]

File: sample-android/templates/initclass_preamble.kte (hunk 5 above)
Comment (ghost) [reply sample-android/templates/initclass_preamble.kte]: Noted. 
I will do so

**Comment by villevoutilainen** [comment]
Excellent, thank you very much!
